src/gl_draw.cxx

```diff
@@ -340,6 +340,11 @@ void gl_texture_fifo::display_texture(int rank)
   // GL_ENABLE_BIT for GL_DEPTH_TEST, GL_LIGHTING
   // GL_TEXTURE_BIT for GL_TEXTURE_RECTANGLE_ARB
   // GL_COLOR_BUFFER_BIT for GL_BLEND and glBlendFunc,
+  // check if the raster pos is valid.
```

Comment:
1. If this patch **would** be correct (see below) it should be moved up **above** all the preceding comments which are related to the following statement.
2. However, since `display_texture` is called only once it should IMHO be moved "up" to the callee, i.e. at the beginning of `Fl_Gl_Window_Driver::draw_string_with_texture` before the texture is computed at all. This would avoid to compute textures that are never drawn.
3. See more comments in the "conversation" tab.